ios/FileUtils.m

```diff
@@ -77,29 +81,80 @@ @implementation FileUtils
+    if(![path hasPrefix:@"file:///"]) {
+        PHAsset* asset = [PHAsset fetchAssetsWithLocalIdentifiers:@[path] options:nil].firstObject;
+        PHContentEditingInputRequestOptions *editOptions = [[PHContentEditingInputRequestOptions alloc]init];
+        editOptions.networkAccessAllowed = YES;
+        
+        // If image, use exif data
+        if ([type isEqualToString:@"image"]) {
+            [asset requestContentEditingInputWithOptions:editOptions completionHandler:^(PHContentEditingInput *contentEditingInput, NSDictionary *info) {
+                CIImage *image = [CIImage imageWithContentsOfURL:contentEditingInput.fullSizeImageURL];
+                NSDictionary *properties = image.properties;
+                NSDictionary *exif = [properties objectForKey:(NSString *)kCGImagePropertyExifDictionary];
+                NSDictionary *datetime = [exif objectForKey:(NSString *)kCGImagePropertyExifDateTimeOriginal];
+                resolve(datetime);
+                return;
+            }];
+        
+        // If not an image, get last modified date
+        } else {
+            resolve(asset.creationDate);
+            return;
+        }
 
-@end
+    // Path for getting exif from file path if image or creation date for video
+    } else {
+        NSString *prefixToRemove = @"file:///";
+        NSString *pathWithoutFilePrefix = [path copy];
+        if ([path hasPrefix:prefixToRemove])
+            pathWithoutFilePrefix = [path substringFromIndex:[prefixToRemove length]];
+        
+        // If image, use exif data
+        if ([type isEqualToString:@"image"]) {
+            NSData* fileData = [NSData dataWithContentsOfFile:pathWithoutFilePrefix];
+            CGImageSourceRef mySourceRef = CGImageSourceCreateWithData((CFDataRef)fileData, NULL);
+            if (mySourceRef != NULL)
+            {
+                NSDictionary *properties = (__bridge NSDictionary *)CGImageSourceCopyPropertiesAtIndex(mySourceRef,0,NULL);
+                NSDictionary *exif = [properties objectForKey:(NSString *)kCGImagePropertyExifDictionary];
+                NSDictionary *datetime = [exif objectForKey:(NSString *)kCGImagePropertyExifDateTimeOriginal];
+                resolve(datetime);
+                return;
+            }
+            
+        // If not an image, get last modified date
+        } else {
+            NSError *error = nil;
+            NSURL *referenceUrl = [NSURL URLWithString:path];
+            NSDate *fileDate;
+            [referenceUrl getResourceValue:&fileDate forKey:NSURLContentModificationDateKey error:&error];
+            if (!error)
+            {
+                NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+                [dateFormatter setDateFormat:@"yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"];
+                NSString *dateString = [dateFormatter stringFromDate:fileDate];
+                
+                resolve(dateString);
+                return;
+            }
```

Comment:
If there is an error, do we want to throw a `reject` at this point?

Reply:
Yes, added!